e2e/deploy_test.go

```diff
@@ -11,57 +11,80 @@ import (
 	"testing"
 
 	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func setup(t *testing.T) *kubernetes.Clientset {
-	kubeConfigVar := os.Getenv("KUBECONFIG")
+var (
+	namespace     = os.Getenv("NAMESPACE")
+	kubeConfig    = os.Getenv("KUBECONFIG")
+	daemonsetName = os.Getenv("DAEMONSET_NAME")
+)
+
+func getKubeConfig(t *testing.T) (*rest.Config, error) {
 	var config *rest.Config
 	var err error
-	if kubeConfigVar != "" {
+	if kubeConfig != "" {
 		t.Logf("No kubeconfig")
-		config, err = clientcmd.BuildConfigFromFlags("", kubeConfigVar)
+		config, err = clientcmd.BuildConfigFromFlags("", kubeConfig)
 	} else {
 		t.Logf("Using ~/.kube/config")
 		config, err = clientcmd.BuildConfigFromFlags("", path.Join(os.Getenv("HOME"), ".kube", "config"))
 	}
 	if err != nil {
-		t.Errorf("Error creating rest config: %v", err)
+		t.Errorf("Error creating rest config")
+		return nil, err
 	}
-	c := kubernetes.NewForConfigOrDie(config)
-	_, err = c.CoreV1().Namespaces().Create(&corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "k8s-image-puller",
-		},
-	})
+	return config, nil
+}
+
+func getClientset(t *testing.T) (*kubernetes.Clientset, error) {
+	config, err := getKubeConfig(t)
 	if err != nil {
-		t.Errorf("Error creating namespace %v", err)
+		return nil, err
 	}
-	go singleCluster.CacheImages()
-	return c
+
+	c := kubernetes.NewForConfigOrDie(config)
+	return c, nil
 }
 
-func teardown(clientset *kubernetes.Clientset, t *testing.T) {
-	utils.DeleteDaemonsetIfExists(clientset)
-	err := clientset.CoreV1().Namespaces().Delete("k8s-image-puller", metav1.NewDeleteOptions(30))
+func TestSingleClusterCacheImages(t *testing.T) {
+
+	type testCase struct {
+		name   string
+		images string
+	}
+
+	testCases := []testCase{
+		{name: "test images", images: "che-theia=quay.io/eclipse/che-theia:next;che-plugin-registry=quay.io/eclipse/che-plugin-registry:next"},
+		{name: "test scratch image", images: "che-machine-exec=quay.io/eclipse/che-machine-exec:next;"},
+		{name: "test volume mount image", images: "volume-mount-image=quay.io/dkwon17/volume-mount:latest"},
```

Comment:
I wasn't quite sure what image to use here, so for now, I put my own image I created with this Dockerfile:
```
FROM alpine
RUN mkdir /volume
RUN echo "hello world" > /volume/greeting
VOLUME /volume
```

Reply:
looks good :+1: